### ISSUE TITLE
GET petitions with ?

### ISSUE THREAD

**Issue by adomingd** [freshOS/Networking#38]
Hello!
I'm testing this library (really nice btw) and I'm with some APIs from https://sampleapis.com/, but when I'm trying to get any list, the debug information prints a server error and the curl prints with a '?' at the final of the path.

Am I doing something wrong? Or why adds a '?' at the end?

This is my code:

```swift
class NetworkCall {
    var network: NetworkingClient = {
        var network = NetworkingClient(baseURL: "https://api.sampleapis.com")
        network.logLevels = .debug
        return network
    }()
    
    func getCoffees() -> AnyPublisher<Coffee, Error> {
        network.get("/coffee/hot")
    }
}
```

And the error:

<img width="1042" alt="image" src="https://user-images.githubusercontent.com/79547771/142684176-4d59041d-7a39-4813-8766-1cf1b80686a3.png">

**Comment by workingDog** [comment]
sorry `adomingd` I did not see your issue sooner.

There is an error in `NetworkingRequest` `private func getURLWithParams()`.
When there are no parameters for the url, an "empty" component is added, 
resulting in having a "?" at the end of the url string.

To fix this in `NetworkingRequest` use:

    private func getURLWithParams() -> String {
        let urlString = baseURL + route
        guard let url = URL(string: urlString) else {
            return urlString
        }
        
        // if no params, do not add any URLComponents
        if params.isEmpty { return urlString }    // <--- here
        
        if var urlComponents = URLComponents(url: url ,resolvingAgainstBaseURL: false) {
            var queryItems = urlComponents.queryItems ?? [URLQueryItem]()
            params.forEach { param in
                // arrayParam[] syntax
                if let array = param.value as? [Any] {
                    array.forEach {
                        queryItems.append(URLQueryItem(name: "\(param.key)[]", value: "\($0)"))
                    }
                }
                queryItems.append(URLQueryItem(name: param.key, value: "\(param.value)"))
            }
            urlComponents.queryItems = queryItems
            return urlComponents.url?.absoluteString ?? urlString
        }
        return urlString
    }

**Comment by s4cha** [comment]
@workingDog Thanks for the fix, pushing it ASAP

**Comment by s4cha** [comment]
@adomingd Can you confirm this is now fixed in [0.3.5](https://github.com/freshOS/Networking/releases/tag/0.3.5) ?

**Comment by adomingd** [comment]
It's fixed now. Thank you much! @workingDog, @s4cha
